plugins/easyimage/plugin.js

```diff
@@ -94,32 +94,37 @@
 	}
 
 	function registerWidget( editor ) {
-		// Overwrite some image2 defaults.
-		editor.on( 'widgetDefinition', function( evt ) {
-			var oldInit = evt.data.init;
+		var config = editor.config;
 
-			if ( evt.data.name !== 'image' ) {
-				return;
-			}
+		CKEDITOR.plugins.imagebase.addImageWidget( editor, 'easyimage', {
+			allowedContent: {
+				figure: {
+					classes: '!' + config.easyimage_class + ',' + config.easyimage_sideClass
+				}
+			},
 
-			evt.data.allowedContent.figure.classes += ',!' + editor.config.easyimage_class + ',' +
-				editor.config.easyimage_sideClass;
+			requiredContent: 'figure(!' + config.easyimage_class + ')',
```

Comment:
Similarly like imagebase, this widget should also require `img[src]` otherwise it has little sense.

This should be covered with small unit test, e.g. in `config.js` suite, just create extra editor with `config.disallowContent='img'`, and ensure that widgets are not upcasted.